src/loader.js

```diff
@@ -48,408 +48,67 @@ export function createLegacyLoaderElement(doc, elementName) {
 }
 
 /* NEW LOADER */
+/** @type {?Promise<!../extensions/amp-loader/0.1/amp-loader.LoaderService>} */
+let loaderServicePromise = null;
+
+/**
+ * Gets a Promise for the LoaderService, initiating a request to download the
+ * code.
+ * @param {!./service/ampdoc-impl.AmpDoc} ampDoc
+ * @param {!Element} element
+ * @return {!Promise<!../extensions/amp-loader/0.1/amp-loader.LoaderService>}
+ */
+function getLoaderServicePromise(ampDoc, element) {
```

Comment:
Have we investigated adding a transform to the AMP Cache to add the amp-loader script to the document when it's served, to remove the delay from when v0.js loads and reaches this point and requests the loader extension?

Reply:
I'll add this to the original issue. My original thinking was that it might be better to not include it, so that the code to load other extensions loads / runs sooner without the loaders code getting in the way.

Reply:
Interesting yeah I guess it's an optimization/UX question, if we want it to be implicitly delayed if that would improve other performance metrics. We would want to state and measure those values we care about when we did this